VueExcelXlsx.vue

```diff
@@ -6,68 +6,58 @@
 
 <script>
     import XLSX from 'xlsx/xlsx';
-    window.$ = window.jQuery = require('jquery');
 
     export default {
         name: "vue-excel-xlsx",
 
         props: {
-            columns: {
+            sheets: {
                 type: Array,
-                default: []
-            },
-            data: {
-                type: Array,
-                default: []
+                default: () => {
+                    return [ { name: 'SheetName', columns: [], data: [] } ]
+                }
             },
             filename: {
                 type: String,
                 default: 'excel'
-            },
-            sheetname: {
-                type: String,
-                default: 'SheetName'
             }
         },
 
         methods: {
             exportExcel() {
-                let createXLSLFormatObj = [];
-                let newXlsHeader = [];
-                let vm = this;
-                if (vm.columns.length === 0){
-                    console.log("Add columns!");
-                    return;
-                }
-                if (vm.data.length === 0){
-                    console.log("Add data!");
-                    return;
-                }
-                $.each(vm.columns, function(index, value) {
-                    newXlsHeader.push(value.label);
-                });
+                let wb = XLSX.utils.book_new()                
 
-                createXLSLFormatObj.push(newXlsHeader);
-                $.each(vm.data, function(index, value) {
-                    let innerRowData = [];
-                    $.each(vm.columns, function(index, val) {
-                        if (val.dataFormat && typeof val.dataFormat === 'function') {
-                            innerRowData.push(val.dataFormat(value[val.field]));
-                        }else {
-                            innerRowData.push(value[val.field]);
-                        }
+                this.sheets.forEach(sheet => {
+                if (sheet.columns.length === 0 || sheet.data.length === 0){
+                    console.log(`Columns or data of sheet '${sheet.name}' missing! Sheet skipped.`);
+                } else {
+                    let createXLSLFormatObj = [];
+                    let newXlsHeader = [];
+                    sheet.columns.forEach(value => {
+                        newXlsHeader.push(value.label);
                     });
-                    createXLSLFormatObj.push(innerRowData);
-                });
 
-                let filename = vm.filename + ".xlsx";
```

Comment:
You forgot to re-add this part, user isn't going to specify the file extension since he's expecting it to be an xlsx.